WordPress/openverse-frontend
# Add VHeaderInternal

## Fixes


Fixes #1807 by @obulat
Fixes #1806 by obulat

## Description


This PR creates a Content header with the links to internal pages. I copied a large part of the implementation from the VFooter PR (#1814).

For this PR, I had to also add the following changes to the `VModal` component:
- Make the VModal controllable from outside by adding the `externalVisible` prop which is used in addition to the `visible` ref inside the component to manage the modal visibility.
- expose the `top-bar` slot from `VModalContent` to the `VModal` consumers.
- add `mode` (light/dark) and `variant` (default/full) props to make the modal content stylable.

## Testing Instructions


1. Run `pnpm dev`.
2. Go to localhost:8443/preferences and check the `new_header` flag
3. Go to localhost:8443/about
4. You should see the new Content header. Check that it works as expected on all breakpoints and matches the [Figma mockups](https://www.figma.com/file/KxkZmBNk218V2Fke1GeYCu/New-header-component?node-id=609%3A70088).

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
